packages/api/cms-api/src/dam/files/file-warning.service.ts

```diff
@@ -0,0 +1,100 @@
+import { EntityManager, FilterQuery } from "@mikro-orm/postgresql";
+import { Inject, Injectable } from "@nestjs/common";
+import { EmitWarningsServiceInterface } from "src/warnings/decorators/emit-warnings.decorator";
+
+import { WarningInput } from "../../warnings/dto/warning.input";
+import { WarningService } from "../../warnings/warning.service";
+import { DamConfig } from "../dam.config";
+import { DAM_CONFIG } from "../dam.constants";
+import { FileInterface } from "./entities/file.entity";
+import { LicenseType } from "./entities/license.embeddable";
+
+@Injectable()
+export class FileWarningService implements EmitWarningsServiceInterface<FileInterface> {
+    constructor(
+        @Inject(DAM_CONFIG) private readonly config: DamConfig,
+        private readonly entityManager: EntityManager,
+        private readonly warningService: WarningService,
+    ) {}
+
+    async emitWarningsBulk() {
+        if (!this.config.enableLicenseFeature) return; // license feature not enabled, no warnings
+
+        const soonToExpireDate = new Date();
+        soonToExpireDate.setDate(soonToExpireDate.getDate() - 30);
+
+        const filterQuery: FilterQuery<FileInterface> = [
+            {
+                license: {
+                    durationTo: {
+                        $ne: soonToExpireDate,
+                    },
+                },
+            },
+        ];
+        if (this.config.requireLicense) {
+            filterQuery.push({ license: null });
+        }
+
+        let files = [];
+        let offset = 0;
+        const limit = 50;
+        do {
+            files = await this.entityManager.getRepository<FileInterface>("DamFile").find(
+                {
+                    $or: filterQuery,
+                },
+                { limit, offset },
+            );
+
+            for (const file of files) {
+                const warnings = await this.emitWarnings(file);
+                await this.warningService.saveWarnings({
+                    warnings,
+                    type: "Entity",
+                    sourceInfo: {
+                        rootEntityName: "DamFile",
+                        rootPrimaryKey: "id",
+                        targetId: file.id,
+                    },
+                });
+            }
+            offset += limit;
+        } while (files.length > 0);
+    }
+
+    async emitWarnings(entity: FileInterface) {
```

Comment:
what does "emit" mean? I know this only from the context of an event emitter pattern.

why not "get"? Or "create"?

Reply:
I agree and I changed it to create for now:
b6f5aae6fafefe550ff967df63c12dc07718195d